test/features/environment.py

```diff
@@ -68,12 +69,27 @@ def after_all(context):
     #        os.environ[k] = v
 
 
+def before_step(context, _):
+    context.atprompt = False
+
+
 def after_scenario(context, _):
     """Cleans up after each test complete."""
 
     if hasattr(context, 'cli') and not context.exit_sent:
-        # Terminate nicely.
-        context.cli.terminate()
+        # Quit nicely.
+        if not context.atprompt:
```

Comment:
Ah ok, I can see what you mean.

@meeuw - do you want to add the `before_scenario` code into this pull request as well since it's so closely related?